src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.cs

```diff
@@ -187,33 +187,33 @@ private void AddGlobalNamespace(INamespaceSymbol globalNamespace)
 
         public override void VisitLocal(ILocalSymbol symbol)
         {
-            if (symbol.IsRef &&
-                format.LocalOptions.IncludesOption(SymbolDisplayLocalOptions.IncludeRef))
+            if (format.LocalOptions.IncludesOption(SymbolDisplayLocalOptions.IncludeModifiers))
             {
-                if (symbol.ScopedKind == ScopedKind.ScopedRef &&
-                    format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.IncludeScoped))
+                if (symbol.IsRef)
                 {
-                    AddKeyword(SyntaxKind.ScopedKeyword);
+                    if (symbol.ScopedKind == ScopedKind.ScopedRef)
+                    {
+                        AddKeyword(SyntaxKind.ScopedKeyword);
+                        AddSpace();
+                    }
+
+                    AddKeyword(SyntaxKind.RefKeyword);
                     AddSpace();
-                }
 
-                AddKeyword(SyntaxKind.RefKeyword);
-                AddSpace();
+                    if (symbol.RefKind == RefKind.RefReadOnly)
+                    {
+                        AddKeyword(SyntaxKind.ReadOnlyKeyword);
+                        AddSpace();
+                    }
+                }
 
-                if (symbol.RefKind == RefKind.RefReadOnly)
+                if (symbol.ScopedKind == ScopedKind.ScopedValue)
                 {
-                    AddKeyword(SyntaxKind.ReadOnlyKeyword);
+                    AddKeyword(SyntaxKind.ScopedKeyword);
                     AddSpace();
                 }
```

Comment:
does this mean you can have `ref scoped` as well as `scoped ref`?

Reply:
> does this mean you can have `ref scoped` as well as `scoped ref`?

The `ILocalSymbol` API allows `ref scoped` but that is not supported from C#.

Reply:
Perhaps we should just make it into an ```else if``` so that `ref` and `non-ref` code paths do not overlap.